src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs

```diff
@@ -80,32 +81,13 @@ internal FunctionInstanceLogger(IFunctionMetadataManager metadataManager, IMetri
 
         public async Task AddAsync(FunctionInstanceLogEntry item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            FunctionInstanceMonitor monitor;
-            item.Properties.TryGetValue(Key, out monitor);
-
-            if (item.EndTime.HasValue)
+            if (item.IsStart)
             {
-                // Function Completed
-                bool success = item.ErrorDetails == null;
-                monitor.End(success);
+                StartFunction(item);
             }
-            else
+            else if (item.IsCompleted)
             {
-                // Function Started
-                if (monitor == null)
-                {
-                    if (_metadataManager.TryGetFunctionMetadata(item.LogName, out FunctionMetadata function))
-                    {
-                        monitor = new FunctionInstanceMonitor(function, _metrics, item.FunctionInstanceId);
```

Comment:
Removing this type completely to remove an object allocation from the invocation path